modules/plugin/epsg-hsql/src/main/java/org/geotools/referencing/factory/epsg/ThreadedHsqlEpsgFactory.java

```diff
@@ -308,7 +308,18 @@ protected AbstractAuthorityFactory createBackingStore(final Hints hints) throws
                 }
             }
         }
-        FactoryUsingHSQL factory = new FactoryUsingHSQL(hints, getDataSource());
+        FactoryUsingHSQL factory =
+                new FactoryUsingHSQL(hints, getDataSource()) {
+                    @Override
+                    protected void shutdown(boolean active) throws SQLException {
```

Comment:
I get that it fixes a test failure, but doing so seems also to break the ability to properly shut down the factory after use. If one disposes the authority factory, will the connection to the database be released? If not, we have a leak.

Reply:
I guess it will, provided that connections are properly disposed of, right?